### PR TITLE
Simulation support: read_only_cache and onchip_memory_cache

### PR DESCRIPTION
**Pull request by artemrad**
# Existing Sample Changes
## Description

Added Sim support

## Type of change
- [X ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [X ] Command Line

### CODE REVIEW

**Comment by artemrad** [comment]
Testing Linux:
* onchip_memory_cache:  https://psg-sc-arc.sc.intel.com/arc/dashboard/reports/show_job/635667787
* read_only_cache:  https://psg-sc-arc.sc.intel.com/arc/dashboard/reports/show_job/640203799

Testing Windows:
* onchip_memory_cache:  https://psg-sc-arc.sc.intel.com/arc/dashboard/reports/show_job/636032931
* read_only_cache:  https://psg-sc-arc.sc.intel.com/arc/dashboard/reports/show_job/640215885

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/onchip_memory_cache/src/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +77,23 @@ set_target_properties(${FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${HARDWARE_C
...
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -Xssimulation -DFPGA_SIMULATOR -Xstarget=<FPGA_DEVICE> onchip_memory_cache.cpp -o onchip_memory_cache.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -Xssimulation -DFPGA_SIMULATOR -o onchip_memory_cache.cpp.o -c onchip_memory_cache.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xstarget=<FPGA_DEVICE> onchip_memory_cache.cpp.o -o onchip_memory_cache.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
+target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/onchip_memory_cache/src/CMakeLists.txt:89]: This line is written twice

File: DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/CMakeLists.txt [hunk 2]
```diff
@@ -63,6 +66,20 @@ set_target_properties(${FPGA_EARLY_IMAGE_CACHE_ENABLED} PROPERTIES COMPILE_FLAGS
...
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -Xssimulation -DFPGA_SIMULATOR -Xstarget=<FPGA_DEVICE> read_only_cache.cpp -o read_only_cache.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -Xssimulation -DFPGA_SIMULATOR -o read_only_cache.cpp.o -c read_only_cache.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xstarget=<FPGA_DEVICE> read_only_cache.cpp.o -o read_only_cache.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
+target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
+add_custom_target(fpga_sim DEPENDS ${SIMULATOR_TARGET} ${SIMULATOR_TARGET})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/CMakeLists.txt:79]:
```suggestion
add_custom_target(fpga_sim DEPENDS ${SIMULATOR_TARGET})
```

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/onchip_memory_cache/src/CMakeLists.txt (hunk 1 above)
Comment (artemrad) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/onchip_memory_cache/src/CMakeLists.txt:89]:
```suggestion
```

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**